### PR TITLE
Support Xcode 16.2 `onPreferenceChange` SDK change

### PR DESCRIPTION
**Pull request by philippzagar**
# Support Xcode 16.2 `onPreferenceChange` SDK change

## :recycle: Current situation & Problem
As shown in https://github.com/StanfordSpezi/SpeziViews/pull/47 and https://github.com/StanfordSpezi/SpeziLLM/issues/76, Apple chose to change the signature of the `onPreferenceChange` view modifier in SwiftUI with the latest Xcode 16.2 RC (also the betas) and the iOS 18.2 SDKs.
In its current state, this prevents the SpeziViews package from compiling in these environments.

The `onPreferenceChange` view modfier now takes a `@Sendable` closure, therefore we cannot capture `@MainActor` isolated properties on the `View` directly anymore: https://developer.apple.com/documentation/swiftui/view/onpreferencechange(_:perform:)?changes=latest_minor
However, as the `@Sendable` closure is still run on the MainActor's serial executor (at least in my testing on 18.2 RC SDKs), we can use `MainActor.assumeIsolated` to avoid scheduling a `MainActor` `Task`, which could delay execution and cause unexpected UI behavior.

## :gear: Release Notes 
- Support Xcode 16.2 `onPreferenceChange` SDK change
- Use Swift 6 language mode in the UITest app

## :books: Documentation
Properly documented in-line why this change was necessary

## :white_check_mark: Testing
Tested the changes locally on Xcode 16.2 RC as well as iOS 18.2 betas.
We should lifting the CI builds / testing to these versions as well, as soon as they're out (should only be a couple of days).

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by philippzagar** [comment]
There's also a concurrency warning in `ReceiveValidationModifier`, as the `ValidationState.Binding` (a non-`Sendable` type) is used in the (now) `@Sendable` closure of `.onPreferenceChange()`.
An easy fix would be to annotate `ValidationState.Binding` as `@unchecked Sendable`, but that's technically no correct as the nested `ValidationContext` cannot be easily made `Sendable` in its current form.
Looking forward to your opinion @Supereg!

File: Sources/SpeziViews/Views/Layout/HorizontalGeometryReader.swift [hunk 1]
```diff
@@ -39,7 +39,13 @@ public struct HorizontalGeometryReader<Content: View>: View {
                 }
             )
             .onPreferenceChange(WidthPreferenceKey.self) { width in
-                self.width = width
+                // The `onPreferenceChange` view modfier now takes a `@Sendable` closure, therefore we cannot capture `@MainActor` isolated properties
+                // on the `View` directly anymore: https://developer.apple.com/documentation/swiftui/view/onpreferencechange(_:perform:)?changes=latest_minor
+                // However, as the `@Sendable` closure is still run on the MainActor (at least in my testing on 18.2 RC SDKs), we can use `MainActor.assumeIsolated`
+                // to avoid scheduling a `MainActor` `Task`, which could delay execution and cause unexpected UI behavior.
+                MainActor.assumeIsolated {
```
Comment (lukaskollmer) [inline Sources/SpeziViews/Views/Layout/HorizontalGeometryReader.swift:46]: if you really wanna be safe you might instead want to do

```swift
if Thread.isMainThread {
    MainActor.assumeIsolated {
        self.width = width
    }
} else {
    Task { @MainActor in
        self.width = width
    }
}
```

that way at least it won't crash should they decide to move it off the main actor in a future release, without documenting that change

Comment (philippzagar) [reply Sources/SpeziViews/Views/Layout/HorizontalGeometryReader.swift:46]: I like that solution @lukaskollmer, gives us way more safety, let's go for that one!

**Comment by philippzagar** [comment]
@Supereg Please do a quick review of the PR, would be great to get this code in soon as the Xcode 16.2 RC was released nearly a week ago (Dec 5), meaning that the public release will probably come in a couple of days.
As lots of Spezi packages and apps depend on `SpeziViews`, it would be great to do this before the public release 🚀

**Review by Supereg — APPROVED** [review]

**Review by PSchmiedmayer — APPROVED** [review]

**Comment by philippzagar** [comment]
@PSchmiedmayer Please merge the PR with bypassing merge requirements. Minor decrease in test coverage as we added a "safety" case in case SwiftUI ever moves the `onPreferenceChange()` closure off the MainActor.

**Comment by PSchmiedmayer** [comment]
Amazing, thank you for the fix @philippzagar!

**Pull request merged**